neutrons/SNAPRed
# implement, test, comment, ewm 4795 apply normalization

## Description of work

This implements the recipe for Applying Normalization in Reduction. 

## Explanation of work

* Supplies the branching logic for applying a normalization if it exists
* Throws an exception if a bg is supplied, not implemented for the next release.
* Organizes the recipe in reusable batchable parts.
* Works around the progress tracking issue by using Utensils, which can be shared.

## To test

Unit tests pass.
When all pieces are in place we should have an integration test.

[EWM#4795](https://ornlrse.clm.ibmcloud.com/ccm/web/projects/Neutron%20Data%20Project%20(Change%20Management)#action=com.ibm.team.workitem.viewWorkItem&id=4795)
